commit 742d32ae157946c40b51b2e8cf3484dea7997e50
Author: Jan Blunck <jblunck@infradead.org>
Date:   Thu Jul 9 10:29:39 2015 +0200

    team: set IFF_SLAVE on team ports
    
    The code in net/ipv6/addrconf.c:addrconf_notify() tests for IFF_SLAVE to
    decide if it should start the address configuration. Since team ports
    shouldn't get link-local addresses assigned lets set IFF_SLAVE when linking
    a port to the team master.
    
    Signed-off-by: Jan Blunck <jblunck@infradead.org>

---
 drivers/net/team/team.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/team/team.c
+++ b/drivers/net/team/team.c
@@ -1122,6 +1122,7 @@ static int team_upper_dev_link(struct te
 	if (err)
 		return err;
 	port->dev->priv_flags |= IFF_TEAM_PORT;
+	port->dev->flags |= IFF_SLAVE;
 	return 0;
 }
 
@@ -1129,6 +1130,7 @@ static void team_upper_dev_unlink(struct
 {
 	netdev_upper_dev_unlink(port->dev, team->dev);
 	port->dev->priv_flags &= ~IFF_TEAM_PORT;
+	port->dev->flags &= ~IFF_SLAVE;
 }
 
 static void __team_port_change_port_added(struct team_port *port, bool linkup);
